test/performance/range/gap_decorator_rand_read_benchmark.cpp

```diff
@@ -0,0 +1,83 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <random>
+#include <utility>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alignment/exception.hpp>
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/decorator/all.hpp>
+#include <seqan3/std/ranges>
+
+#include "gap_decorator_helper.hpp"
+
+using namespace seqan3;
+
+// ============================================================================
+//  read at random position
+// ============================================================================
+template <typename gap_decorator_t, bool gapped_flag>
+void read_random(benchmark::State& state)
+{
+    unsigned int seq_len = state.range(0);
+    using size_type = typename gap_decorator_t::size_type;
+    using sequence_type = remove_cvref_t<detail::unaligned_seq_t<gap_decorator_t>>;
+    sequence_type seq(seq_len, 'A'_dna4);
+
+    // vector of sampled gap lengths for each position
+    std::vector<size_type> gaps(seq_len, 0);
+
+    // determine sum of gaps and non-gap symbols for not exceeding targeted sequence length
+    if constexpr(gapped_flag)
+    {
+        sample<size_type>(gaps, seq_len, state.range(1)/100.0);
+        resize<size_type, sequence_type>(gaps, seq, seq_len);
+    }
+
+    // initialize with (truncated) sequence and insert gaps from left to right
+    gap_decorator_t gap_decorator;
+    assign_unaligned(gap_decorator, seq);
+
+    // insert gaps before starting benchmark
+    if constexpr(gapped_flag)
+        insert_gaps<gap_decorator_t>(gaps, gap_decorator, seq_len);
+
+    std::mt19937 generator(time(0)); //Standard mersenne_twister_engine seeded with current time
+    std::uniform_real_distribution<> uni_dis{0.0, static_cast<double>(seq_len)};
+
+    // sample read positions in advance
+    std::vector<size_t> access_positions;
+    access_positions.resize(SEQ_LEN_SHORT);
+    #ifdef SEQAN3_LONG_TESTS
+        access_positions.resize(SEQ_LEN_LONG)
+    #endif
+
+    for (size_t i = 0; i < access_positions.size(); ++i)
+        access_positions[i] = uni_dis(generator);
+    size_t j = 0;
+    for (auto _ : state)
+    {
+        benchmark::DoNotOptimize(gap_decorator[access_positions[j++]]);
+        if (j == access_positions.size())
+            j = 0;
```

Comment:
I expect the if check to be a lot more expensive then the look-up. Can you ensure that j is divisable by 10 and perform 10 random access lookups ever loop iteration?

Reply:
I made this benchmark consistent with the other one in the same file, so I use `j %= 1 << 10` instead of if-clause.

Reply:
Do you mean 10 random access ops per state iteration, right?

Reply:
> Do you mean 10 random access ops per state iteration, right?

Yes!

Reply:
I implemented this change, but it obfuscates a bit what is tested in the benchmark and makes  less comparable with other benchmarks.